src/dlls/mscorpe/pewriter.cpp

```diff
@@ -2026,7 +2028,11 @@ HRESULT PEWriter::fixup(CeeGenTokenMapper *pMapper)
                         {
                             IMAGE_SECTION_HEADER* phdr = textSection->m_header;
                             // Add to reloc table
+#ifdef FEATURE_PAL
+                            ir.u.VirtualAddress = VAL32(rva);
```

Comment:
Could you please rather change the IMAGE_RELOCATION to use anonymous union? Then you can use the same macro as for Windows. Anonymous unions are now allowed both in C and C++ (in C starting from C11 which we set in the compiler options)